### ISSUE TITLE
route transform event streams cannot be referenced using wildcards

### ISSUE THREAD

**Issue by jeromekleinen-kbc** [timberio/vector#7605]
### Vector Version

```
vector 0.13.1 (v0.13.1 x86_64-apple-darwin 2021-04-29)
```

### Vector Configuration File

```toml
[sources.in]
type = "generator"
format = "syslog"

[transforms.route]
type = "route"
inputs = ["in"]

route.a = '''contains(.message, "a")'''
route.b = '''contains(.message, "b")'''

[transforms.remap]
type = "remap"
inputs = ["route.*"]
source = """
.foo = "bar"
"""

[sinks.out]
type = "console"
inputs = ["remap"]
encoding.codec = "json"
```

### Expected Behavior

The event streams coming out of the route transform should be able to be referenced using the wildcard notation.

### Actual Behavior

vector won't start with the following error:

```
vector % vector -c route.toml
May 26 21:12:29.453  INFO vector::app: Log level is enabled. level="vector=info,codec=info,vrl=info,file_source=info,tower_limit=trace,rdkafka=info"
May 26 21:12:29.453  INFO vector::app: Loading configs. path=[("route.toml", None)]
May 26 21:12:29.461 ERROR vector::cli: Configuration error. error=Transform "remap" has no inputs
```

**Comment by lukesteensen** [comment]
The naive fix would just be to move macro expansion before wildcard expansion here:

https://github.com/timberio/vector/blob/65671bead5251d8c7be65427a01299ce3e3e4ac8/src/config/compiler.rs#L7-L9

We should be careful to think through any weird dependencies between the two phases though. Off the top of my head, I can't think of any issues that would cause.

**Comment by jszwedko** [comment]
I did exactly that fix in #6786 to switch the ordering.
